daemon/fqdn.go

```diff
@@ -241,6 +248,9 @@ func (d *Daemon) bootstrapFQDN(restoredEndpoints *endpointRestoreState) (err err
 				if err != nil {
 					log.WithError(err).Error("error updating internal DNS cache for rule generation")
 				}
+
+				metrics.ProxyUpstreamTime.WithLabelValues("", metrics.L7DNS).Observe(
```

Comment:
It seems that there's a few error conditions above this statement here which are not appropriately handled, where an error is returned without updating the metric. Why not put this particular case with the others in the switch statement above so we can ensure that around that part of the code, all paths attribute the proxyupstreamtime? (ie allowed/!allowed cases)

Reply:
There is two calls to the same function here: 

1) In case of error, there is no response at all. So we need to trigger that on the start of the function. 
2) If is a response, we need to trigger the correct time, on the second call to the callback and just in the case that it's the response. 

REgards.

Reply:
Is still the upstream no? The TotalTime we don't have it at all because is not received in the callback.

Reply:
Sorry, I didn't see the responses.

It seems like the `endMetric()` callback solves my concern above.